PyDSS/pyControllers/Controllers/MotorStallSimple.py

```diff
@@ -0,0 +1,102 @@
+        self._ControlledElm = MotorObj
+        self.__Settings = Settings
+        self.__dssSolver = dssSolver
+
+        self._ControlledElm.SetParameter('model', 2)
+        self._ControlledElm.SetParameter('vminpu', 0.0)
+        # self.kw = self._ControlledElm.GetParameter('kw')
+        # self.kvar = self._ControlledElm.GetParameter('kvar')
+        self.kw = self.__Settings['ratedKW']
+        S = self.kw / self.__Settings['ratedPF']
+        self.kvar = math.sqrt(S**2 - self.kw**2)
+        self._ControlledElm.SetParameter('kw', self.kw)
+        self._ControlledElm.SetParameter('kvar', self.kvar)
+        self.stall_time_start = 0
+        self.stall = False
+        self.disconnected =False
+        self.Tdisconnect_start = 0
+        return
+
+
+    def Name(self):
+        return self.name
+
+    def ControlledElement(self):
+        return "{}.{}".format(self._class, self._name)
+
+    def debugInfo(self):
+        return [self.__Settings['Control{}'.format(i+1)] for i in range(3)]
+
+
+    def Update(self, Priority, Time, UpdateResults):
+        if Priority == 0:
+            Vbase = self._ControlledElm.sBus[0].GetVariable('kVBase')
+            Ve_mags = max(self._ControlledElm.GetVariable('VoltagesMagAng')[::2])/ 120.0
+
+
+            if Ve_mags < self.__Settings['Vstall'] and not self.stall:
+                print(Ve_mags)
+                self._ControlledElm.SetParameter('kw', self.kw * self.__Settings['Pfault'] )
+                self._ControlledElm.SetParameter('kvar', self.kw * self.__Settings['Qfault'] )
+                self._ControlledElm.SetParameter('model', 1)
+                self.stall = True
+                self.stall_time_start = self.__dssSolver.GetTotalSeconds()
+                return 0.1
+            return 0
+        if Priority == 1:
+            if self.stall:
+                self.stall_time = self.__dssSolver.GetTotalSeconds() - self.stall_time_start
+                #print(self.stall_time)
+                if self.stall_time > self.__Settings['Tprotection']:
+                    self.stall = False
+                    self.disconnected = True
+                    self._ControlledElm.SetParameter('kw', 0)
+                    self._ControlledElm.SetParameter('kvar', 0)
+                    self.Tdisconnect_start = self.__dssSolver.GetTotalSeconds()
+                return 0 #self.model_1(Priority)
+            return 0
+        if Priority == 2:
+            if self.disconnected:
+                time = self.__dssSolver.GetTotalSeconds() - self.Tdisconnect_start
+                if time > self.__Settings['Treconnect']:
+                    self.disconnected = False
+                    self._ControlledElm.SetParameter('kw', self.kw)
+                    self._ControlledElm.SetParameter('kvar', self.kvar)
+                    self._ControlledElm.SetParameter('model', 2)
+                    self._ControlledElm.SetParameter('vminpu', 0.0)
+
```

Comment:
Should it raise an exception if `Priority` is something else?

Reply:
no this implementation is correct. It has been validated multiple times.

Reply:
The user could set Priority to 3 and this code won’t complain; it just won’t do anything. If that is your desired behavior, ok.

Reply:
added an assertion